File: backend/src/main/java/fr/irvineflth/EcoFleet/dto/VehicleDto.java
package fr.irvineflth.EcoFleet.dto;

import fr.irvineflth.EcoFleet.domain.enums.VehicleType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class VehicleDto {
    private Long id;
    private String registrationNumber;
    private VehicleType type;
    private String brand;
    private String model;
    private Integer autonomyKm;
    private Integer currentKm;
    private LocalDate lastMaintenanceDate;
    private boolean available;
}

File: frontend/src/app/core/dto/vehicle.dto.ts
import {VehicleType} from '../models/vehicle-type.enum';

export interface VehicleDto {
  registrationNumber: string;
  type: VehicleType;
  brand: string;
  model: string;
  autonomyKm: number;
  currentKm: number;
  available: boolean;
}
